src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1554,6 +1560,32 @@ internal static void ApplyConfiguration(JObject config, ScriptHostConfiguration
                 }
             }
 
+            JObject languageWorkerSection = (JObject)config["languageWorker"];
+            value = null;
+            if (languageWorkerSection != null)
+            {
+                if (languageWorkerSection.TryGetValue("maxMessageLength", out value))
+                {
+                    var requestedGrpcMaxMessageLength = int.Parse((string)value);
+                    if (ScriptSettingsManager.Instance.IsDynamicSku)
+                    {
+                        string message = $"Cannot set {nameof(scriptConfig.MaxMessageLength)} on Consumption plan. Default MaxMessageLength: {DefaultMaxMessageLengthDynamicSku} will be used";
+                        logger?.LogWarning(message);
+                    }
+                    else
+                    {
+                        scriptConfig.MaxMessageLength = requestedGrpcMaxMessageLength * 1024 * 1024;
+                        if (scriptConfig.MaxMessageLength < 0)
+                        {
+                            // Current grpc max message limits
+                            string message = $"MaxMessageLength must be between 4MB and 2000MB.Default MaxMessageLength: {DefaultMaxMessageLength} will be used";
+                            logger?.LogWarning(message);
+                        }
+                        scriptConfig.MaxMessageLength = scriptConfig.MaxMessageLength > 0 ? scriptConfig.MaxMessageLength : DefaultMaxMessageLength;
```

Comment:
It looks like you're making zero mean "default" here. If you want these semantics, maybe make it nullable and then this line can could change into `scriptConfig.MaxMessageLength = scriptConfig.MaxMessageLength ?? Default`.

Alternatively, consider a design where this method does not update `scriptConfig.MaxMessageLength` multiple times. Might be better to have a local variable that it updates as it goes and then finally sets `scriptConfig.MaxMessageLength` when its done.